src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/MSITest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.AlwaysEncrypted;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+import com.microsoft.sqlserver.jdbc.SQLServerDataSource;
+import com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement;
+import com.microsoft.sqlserver.jdbc.SQLServerResultSet;
+import com.microsoft.sqlserver.jdbc.SQLServerStatement;
+import com.microsoft.sqlserver.jdbc.TestResource;
+import com.microsoft.sqlserver.jdbc.TestUtils;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Constants;
+import com.microsoft.sqlserver.testframework.PrepUtil;
+
+
+/**
+ * Tests involving MSI authentication
+ */
+@RunWith(JUnitPlatform.class)
+@Tag(Constants.MSI)
+public class MSITest extends AESetup {
+    /*
+     * Test basic MSI auth
+     */
+    @Test
+    public void testAuth() throws SQLException {
+        try (SQLServerConnection con = PrepUtil.getConnection(connectionString)) {} catch (Exception e) {
+            fail(TestResource.getResource("R_loginFailed") + e.getMessage());
+        }
+    }
+
+    /*
+     * Test MSI auth using datasource
+     */
+    @Test
+    public void testDSAuth() throws SQLException {
+        SQLServerDataSource ds = new SQLServerDataSource();
+        AbstractTest.updateDataSource(connectionString, ds);
+
+        try (Connection con = ds.getConnection(); Statement stmt = con.createStatement()) {} catch (Exception e) {
+            fail(TestResource.getResource("R_loginFailed") + e.getMessage());
+        }
+    }
+
+    /*
+     * Test AKV with with credentials
+     */
+    @Test
+    public void testCharAkvWithCred() throws SQLException {
+        // unregister the custom providers registered in AESetup
+        SQLServerConnection.unregisterColumnEncryptionKeyStoreProviders();
+
+        // add credentials to connection string
+        AETestConnectionString = TestUtils.addOrOverrideProperty(AETestConnectionString,

```

Comment:
I think currently your testing might not be testing the correct inputs from the user. This AETestConnectionString is a static variable that keeps getting added more parameters as individual tests get executed, resulting in latter tests having the same connection property defined multiple times. You should reset this variable to the base connection string at the beginning of each test run.